docs/modules/ROOT/pages/how-to/use-k8up.adoc

```diff
@@ -0,0 +1,353 @@
+
+To follow this guide, please make sure that you have the following tool installed:
+
+`oc`:: You can download the OpenShift command directly from {product}, selecting the help menu (marked as a question mark) and selecting the "Command line tools" entry.
+
+`restic`:: Follow the https://restic.net/#installation[Restic installation instructions] corresponding to your operating system.
+
+== Procedure
+
+This section explains the steps required to use K8up with {product}.
+
+=== 1. Login to {product}
+
+include::partial$login-in-terminal.adoc[]
+
+. Create a new project called "[YOUR_USERNAME]-fortune-go"
++
+[source,shell]
+--
+oc new-project [YOUR_USERNAME]-fortune-go
+--
+
+=== 2. Deploy a simple app
+
+We're going to deploy a https://gitlab.com/vshn/applications/fortune-go[sample application written in Go] using the `deployment.yaml` file shown below. This deployment doesn't need neither a `Service` nor an `Ingress`:
+
+[source,yaml]
+--
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: fortune-go
+spec:
+  template:
+    spec:
+      containers:
+      - image: registry.gitlab.com/vshn/applications/fortune-go:latest
+        imagePullPolicy: Always
+        name: fortune-container
+        ports:
+        - containerPort: 8080
+    metadata:
+      labels:
+        app: fortune-go
+      annotations:
+        k8up.io/backupcommand: fortune # <1>
+        k8up.io/file-extension: .txt   # <2>
+  selector:
+    matchLabels:
+      app: fortune-go
+  strategy:
+    type: Recreate
+--
+<1> K8up is able to backup the output of any tool piping data to `stdin`; in this case, we're going to backup a message created by the `fortune` tool installed in the pod.
+<2> This annotation provides a file extension to the data to be added in the backup.
+
+Apply the deployment to your {product} project:
+
+[source,shell]
+--
+oc apply -f deployment.yaml
+--
+
+=== 3. Deploy Minio
+
+K8up is powered by the https://restic.net/[Restic] backup tool, and is thus able to store backups in any S3-compatible storage. For the purposes of this guide, you can quickly install a https://min.io/[MinIO] instance in your project.
+
+IMPORTANT: This setup is only for demo purposes. In production settings, it's recommended to use an S3 provider outside of the cluster, such as https://aws.amazon.com/s3/[Amazon S3], https://www.exoscale.com/object-storage/[Exoscale Object Storage], a secure MinIO instance in another {product} zone, or similar.
+
+[source,yaml]
+--
+apiVersion: v1
+kind: Secret  # <1>
+metadata:
+  name: minio-credentials
+type: Opaque
+stringData:
+  username: minio
+  password: minio123
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: minio-pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 20Gi
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: minio
+spec:
+  selector:
+    matchLabels:
+      app: minio
+  strategy:
+    type: Recreate
+  template:
+    metadata:
```

Comment:
Why not install minio with Helm? (https://github.com/minio/minio/tree/master/helm/minio) Saves a lot of manually maintaining these deployment manifests here.
We'd probably need a values.yaml file to configure things like Ingress and credentials

Reply:
Let's use this now and consider the Helm chart for a future version.